### PR TITLE
CBL-817: Add unit tests for URLEndpointListener 

### PR DESCRIPTION
**Pull request by jayahariv**
* add test for validating  ports.
* add tests for get URLs, but test fails due to wrong scheme in URLs
* add tests for connection status, but again fails the active connection count, lite core fails.

### CODE REVIEW

File: Objective-C/Tests/URLEndpointListenerTest.m [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+typedef CBLURLEndpointListener Listener;
+
+@end
+
+@implementation URLEndpointListenerTest
+
+- (Listener*) listenAt: (uint16)port {
+    Config* config = [[Config alloc] initWithDatabase: self.otherDB port: port];
+    config.disableTLS = YES;
+    Listener* list = [[Listener alloc] initWithConfig: config];
```
Comment (pasin) [inline Objective-C/Tests/URLEndpointListenerTest.m:40]: I feel like list is not a good variable name for this as list also means List. Can we change it to listener?

File: Objective-C/Tests/URLEndpointListenerTest.m [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+- (void) testBusyPort {
+    Listener* list1 = [self listenAt: kPort];
+    
+    // initialize a listener at same port
+    Config* config = [[Config alloc] initWithDatabase: self.otherDB port: kPort];
+    config.disableTLS = YES;
+    Listener* list2 = [[Listener alloc] initWithConfig: config];
+    
+    // already in use when starting the second listener
+    [self ignoreException:^{
```
Comment (pasin) [inline Objective-C/Tests/URLEndpointListenerTest.m:95]: Why do we need to ignore the exception here?

File: Objective-C/Tests/URLEndpointListenerTest.m [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    }];
+    
+    // stops
+    [list1 stop];
+}
+
+// TODO: https://issues.couchbase.com/browse/CBL-948
+- (void) _testURLs {
+    Config* config = [[Config alloc] initWithDatabase: self.otherDB port: kPort];
+    config.disableTLS = YES;
```
Comment (pasin) [inline Objective-C/Tests/URLEndpointListenerTest.m:109]: We also need to test without disabling TLS as well.

File: Objective-C/Tests/URLEndpointListenerTest.m (hunk 2 above)
Comment (jayahariv) [reply Objective-C/Tests/URLEndpointListenerTest.m:95]: may be, the exception thrown from lite core might not be caught in between? will check

File: Objective-C/Tests/URLEndpointListenerTest.m (hunk 3 above)
Comment (jayahariv) [reply Objective-C/Tests/URLEndpointListenerTest.m:109]: should we enable TLS for all tests? or only a few to verify?

Comment (pasin) [reply Objective-C/Tests/URLEndpointListenerTest.m:114]: TLS mode is the default mode so all the tests should test with TLS. Ideally, we possible we test both TLS and non-TLS but some may not need such as testPort.

File: Objective-C/Tests/URLEndpointListenerTest.m (hunk 2 above)
Comment (jayahariv) [reply Objective-C/Tests/URLEndpointListenerTest.m:95]: already existing port will throw exception from liter core. so the problem lies in hitting the exception break point

Comment (jayahariv) [reply Objective-C/Tests/URLEndpointListenerTest.m:95]: will keep this as same. and merging, will fix this if any issue exists.

**Pull request merged**